src/Microsoft.OData.Core/UriParser/Parsers/SelectExpandTermParser.cs

```diff
@@ -170,6 +165,8 @@ private PathSegmentToken ParseSegment(PathSegmentToken previousSegment, bool all
                 this.lexer.NextToken();
             }
 
+            // By design, "/$count" or "/$ref" should return using SystemToken, but the existing implementation is using "NonSystemToken".
+            // So far, let's keep it unchanged.

```

Comment:
should we fix this in ODL9?

Reply:
Good question

Reply:
It could be related to another question. What's the purpose of 'SystemToken' and 'NonSystemToken'? From current implementation, it seems there's no difference between them. So, In the ODL9, we could do a refactoring on it.